src/CabanaPD_Solver.hpp

```diff
@@ -461,58 +442,45 @@ class SolverFracture
         // Main timestep loop.
         for ( int step = 1; step <= num_steps; step++ )
         {
+            _step_timer.start();
+
             // Integrate - velocity Verlet first half.
-            integrate_timer.reset();
             integrator->initialHalfStep( *particles );
-            integrate_time += integrate_timer.seconds();
 
             // Update ghost particles.
-            comm_timer.reset();
             comm->gatherDisplacement();
-            comm_time += comm_timer.seconds();
 
             // Compute/communicate LPS weighted volume (does nothing for PMB).
-            force_timer.reset();
             force->computeWeightedVolume( *particles, *neighbors, mu );
-            force_time += force_timer.seconds();
-            comm_timer.reset();
             comm->gatherWeightedVolume();
-            comm_time += comm_timer.seconds();
             // Compute/communicate LPS dilatation (does nothing for PMB).
-            force_timer.reset();
             force->computeDilatation( *particles, *neighbors, mu );
-            force_time += force_timer.seconds();
-            comm_timer.reset();
             comm->gatherDilatation();
-            comm_time += comm_timer.seconds();
 
             // Compute internal forces.
-            force_timer.reset();
             computeForce( *force, *particles, *neighbors, mu,
                           neigh_iter_tag{} );
-            force_time += force_timer.seconds();
 
             // Add boundary condition.
             boundary_condition.apply( exec_space{}, *particles );
 
             // Integrate - velocity Verlet second half.
-            integrate_timer.reset();
             integrator->finalHalfStep( *particles );
-            integrate_time += integrate_timer.seconds();
 
             // Print output.
             if ( step % output_frequency == 0 )
             {
-                energy_timer.reset();
                 auto W = computeEnergy( *force, *particles, *neighbors, mu,
                                         neigh_iter_tag() );
-                energy_time += energy_timer.seconds();
 
-                output_timer.reset();
-                this->step_output( step, W );
                 particles->output( step / output_frequency, step * dt,
                                    output_reference );
-                output_time += output_timer.seconds();
+                _step_timer.stop();
+                this->step_output( step, W );
```

Comment:
Why is this line below the timer?

Reply:
Because the timer is not correct if it's not stopped - this was the issue you previously pointed out